src/librustdoc/clean/types.rs

```diff
@@ -1343,14 +1363,14 @@ impl Type {
         }
     }
 
-    crate fn generics(&self) -> Option<Vec<Type>> {
+    crate fn generics<'a>(&'a self) -> Option<Vec<&'a Type>> {
         match *self {
             ResolvedPath { ref path, .. } => path.segments.last().and_then(|seg| {
                 if let GenericArgs::AngleBracketed { ref args, .. } = seg.args {
                     Some(
                         args.iter()
                             .filter_map(|arg| match arg {
-                                GenericArg::Type(ty) => Some(ty.clone()),
+                                GenericArg::Type(ty) => Some(ty),
```

Comment:
It's part of the commit where I made the `generics` function return a vector of references instead of cloning by default. This change was allowed by my update so since it was linked, I thought it might be nice to do it in the same PR.